UserGuide.md

```diff
@@ -140,6 +140,18 @@ Note that you can not serialize objects with circular references since that will
 // Deserialization
 BagOfPrimitives obj2 = gson.fromJson(json, BagOfPrimitives.class);
 // ==> obj2 is just like obj
+
+// If you the type of your json is an Object you and deserialize into a generic Map class
+Map map = gson.fromJson(json, Map.class);
+// ==> map is: {value1=1.0, value2=abc}
+
+// Or you can just use Object.class, and trust gson to return an Object of the correct type
+Object mapObj = gson.fromJson(json, Object.class);
+// ==> mapObj is a Map
+Object numObj = gson.fromJson("1", Object.class);
+// ==> numObj is a Double
+Object listObj = gson.fromJson("[1, 3, {\"k\":\"v\"}]", Object.class);
+// ==> listObj is a List
```

Comment:
If the user wants a specific type (Map, Double or List) then it is a little bit weird to ask Gson to deserialize an Object. I think this might rather confuse the user.
Additionally it does not match the previous example code where `json` is the JSON representation of `BagOfPrimitives` and therefore **not** a Double or List.

Reply:
My point is that you don't have to use a BagOfPrimitives.  If you just call `gson.fromJson("...", Object.class)` Gson gives you a List.

Reply:
Ah nevermind sorry, I overlooked that you are providing a custom JSON string for these cases.

Though I still think that asking Gson to deserialize an `Object` when you are actually interested in a specific type is a little bit weird and might give a bad impression.
If the user is dealing with JSON of unknown structure, then pointing them to `JsonElement` and related methods might be more appropriate.

Reply:
JsonElement and related objects are hard to figure out how to use.  If you just have one line of `Object obj = gson.fromJson(jsonStr, Object.class)` you can use standard java if statements with instanceof to handle it.

The motivation for this MR is that I do codepairs with job candidates in HackerRank.  The HackerRank java8 platform only includes a few libraries, one of which is Gson.  Most candidates have not used Gson before so they have to quickly look up some documentation how to use it.  Some try writing custom classes to parse into.  Some try using JsonElement.  The ones that try using JsonElement have a hard time because they don't know exactly how to use the Gson classes.  A lot of the time they end up extracting a string with quotes in it, and have to do `str.replaceAll("\"", "")` which would be unforgivable in real code.  I don't want them wasting time figuring out how to use Gson, I want them to be able to quickly figure out how to get minimal correct functionality out of it.  If they can do  `Object obj = gson.fromJson(jsonStr, Object.class)` that's all they need to know.